### ISSUE TITLE
Allow providing base version as argument

### ISSUE THREAD

**Issue by OlivierDupre** [docker/docker#31688]
As shown on in the official doc: https://docs.docker.com/compose/compose-file/#args you can pass arguments from docker-compose to the build process.
But, the Dockerfile must start with directive FROM, and an argument is available only after calling ARG your-arg-name.
So one cannot provide base version through build argument.

Could it be possible to make this working? So we can call something like this: FROM ubuntu:$UBUNTU_VERSION?

**Comment by thaJeztah** [comment]
Thanks for opening this feature request @OlivierDupre. This actually is a duplicate of https://github.com/docker/docker/issues/18119, and there is a pull request implementing this here; https://github.com/docker/docker/pull/31352, which is currently being discussed.

I'll close this issue, because it's a duplicate of https://github.com/docker/docker/issues/18119; you can subscribe to that issue to keep informed. Feel free to comment here after I closed 👍
